eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php

```diff
@@ -349,8 +367,28 @@ public function publish( $typeId )
         // Clear type cache and all identifier cache (as we don't know the identifier)
         $this->cache->clear( 'contentType', $typeId );
         $this->cache->clear( 'contentType', 'identifier' );
+        $this->cache->clear( 'fieldMap' );
 
         // clear content cache
         $this->cache->clear( 'content' );//TIMBER! (possible content changes)
     }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\Type\Handler::getFieldMap
+     */
+    public function getFieldMap( $legacy = false )
```

Comment:
The requirements are a bit different, for Solr and Elasticsearch we need (mainly to get FieldType's `Indexable` implementation from the registry):

```php
[
    "content-type-identifier" => [
        "field-definition-identifier" => "field-type-identifier",
        ...
    ],
    ...
]
```

For Legacy Search Engine it looks the same aside form FieldDefinition ID instead of FieldType identifier. So we don't need to get `Indexable` implementations, but OTOH need to match field table through FieldDefinition ID (`ezcontentobject_attribute`).

My reason for handling the difference in this way is that we planned to eventually remove Legacy Search Engine (or rather any storage-based search engine). Even if it might make sense to keep it as a synchronous search engine for internal usage, Field search should not be a part of it (I'm still not 100% sure on this and we never actually discussed it). So this approach basically keeps the interface and implementation as it should look in the end. The other approach would be to put both FieldType identifier and FieldDefinition ID to the returned field map, and then sort it out when Legacy Search Engine gets out of the picture.